server/middleware/errorMessageMiddleware.ts

```diff
@@ -0,0 +1,18 @@
+import { NextFunction, Request, Response } from 'express'
+
+export default function errorMessageMiddleware(req: Request, res: Response, next: NextFunction) {
+  res.redirectWithErrors = (path: string, messages: Record<string, string>[]): void => {
+    req.flash('errors', JSON.stringify(messages))
+    res.redirect(path)
+  }
+
+  if (req.method === 'GET') {
+    const errorMessages = req.flash('errors')[0]
+
+    if (errorMessages) {
+      res.locals.errors = JSON.parse(errorMessages)
+    }
+  }
+
+  next()
+}
```

Comment:
Assuming we kept the original flash errors declaration in `express/index.d.ts` above, I wonder if we could simplify this by not going via json representations of the array? Something like:
```
export default function errorMessageMiddleware(req: Request, res: Response, next: NextFunction) {
  res.redirectWithErrors = (path: string, errors: Record<string, string>[]): void => {
    req.flash('errors', errors)
    res.redirect(path)
  }

  if (req.method === 'GET') {
    res.locals.errors = req.flash('errors')
  }

  next()
}
```

Would that work? 🤷‍♂️